evm_loader/program/src/executor.rs

```diff
@@ -125,11 +125,11 @@ impl<'a, B: AccountStorage> Handler for Executor<'a, B> {
     }
 
     fn gas_left(&self) -> U256 {
-        U256::one() // TODO
+        self.state.gas_limit().saturating_sub(self.gasometer.used_gas())
```

Comment:
`self.gasometer.used_gas()` is a gas used in a current iteration.
You need to sum gas from all iterations before using it here

Reply:
fixed